cpp/ql/lib/semmle/code/cpp/models/implementations/Printf.qll

```diff
@@ -172,7 +173,7 @@ private class SnprintfImpl extends Snprintf, AliasFunction, SideEffectFunction {
  *      and
  *      https://learn.microsoft.com/en-us/previous-versions/windows/embedded/ms860435(v=msdn.10)
  */
-private class StringCchPrintf extends FormattingFunction {
+private class StringCchPrintf extends FormattingFunction, NonThrowingFunction {
```

Comment:
I think `StringCchPrintf` is [C++](https://learn.microsoft.com/en-us/windows/win32/api/strsafe/nf-strsafe-stringcchprintfa), so if it lacks the `nothrow` tag we've little reason to believe it won't throw?

Reply:
I see. I've removed the extension of `NonThrowingFunction` for `StringCchPrintf`.